mathics/core/expression.py

```diff
@@ -1120,7 +1120,7 @@ def has_symbol(self, symbol_name: str) -> bool:
             if hasattr(element, "has_symbol")
         )
 
-    def restructure(self, head, elements, evaluation, structure_cache=None, deps=None):
+    def restructure(self, head, elements, evaluation, structure_cache={}, deps=None):
```

Comment:
Shouldn't it be better to create the dictionary inside the function instead of setting a mutable element as a default parameter?

Reply:
Ok. I'll change this, and remove the default parameter.